add new parameter documentation

mkrorbrkr opened the pull request:
<!-- If you're unsure about any question, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code. -->

### Attention (CC) @LongForWisdom @prose11 

<!--- Bring to the attention of others -->

## Summary of changes

Added documentation for Surplus lot size and DSR

LongForWisdom commented on governance/parameter-docs/param-dai-savings-rate.md:
````diff
@@ -0,0 +1,53 @@
+```
+
+## Description
+The Dai Savings Rate allows Dai holders to deposit Dai into the contract and earn interest paid in Dai. Maker Governance can alter the rate at which interest accrues.
````
Maybe focus this more on the perspective of governance. For example, the first sentence should refer to governance rather than DAI Holders. "The Dai Savings Rate parameter allows Maker Governance to determine the interest rate available to DAI Holders when locking DAI in the DSR smart contract."

LongForWisdom commented on governance/parameter-docs/param-dai-savings-rate.md:
```diff
@@ -0,0 +1,53 @@
+## Description
+The Dai Savings Rate allows Dai holders to deposit Dai into the contract and earn interest paid in Dai. Maker Governance can alter the rate at which interest accrues.
+
+The Dai Savings rate is usually expressed as a percentage and can be any value greater than 0%. Therefore, the DAI Savings rate cannot be negative.
```
"can be set to any value greater than or equal to 0%". I'd also lose the second sentence, feels superfluous.

LongForWisdom commented on governance/parameter-docs/param-dai-savings-rate.md:
```diff
@@ -0,0 +1,53 @@
+
+The Dai Savings rate is usually expressed as a percentage and can be any value greater than 0%. Therefore, the DAI Savings rate cannot be negative.
+
+The Dai Savings Rate can also be turned off by Maker Governance - rendering the interest rate in effect to be 0%.
```
Is this true? Or do you just mean it can be turned off by setting it to zero percent? Either way, it probably doesn't need to be in the first paragraph.

LongForWisdom commented on governance/parameter-docs/param-dai-savings-rate.md:
```diff
@@ -0,0 +1,53 @@
+The Dai Savings Rate can also be turned off by Maker Governance - rendering the interest rate in effect to be 0%.
+
+## Purpose
+Maker Governance can utilise the Dai Savings Rate to encourage Dai holders to deposit Dai to the Dai Savings Rate contract. Higher amounts of Dai within the Dai Savings Rate
```
Maybe try to be really specific here as to what the purpose is. The purpose is to allow Maker Governance to adjust the level of demand for DAI (by encouraging DAI holders to deposit DAI in the DSR smart contract.) Second sentence is good. except for random line break.

LongForWisdom commented on governance/parameter-docs/param-dai-savings-rate.md:
```diff
@@ -0,0 +1,53 @@
+should reduce the market supply of Dai and act as a means to control the Peg.
+
+## Trade-offs
+In essence, a higher Dai Savings Rate results in upward pressure on Dai price, and a lower Dai Savings Rate will reduce this pressure.
```
This maybe first better in the next section (on why it should be changed)

LongForWisdom commented on governance/parameter-docs/param-dai-savings-rate.md:
```diff
@@ -0,0 +1,53 @@
+## Trade-offs
+In essence, a higher Dai Savings Rate results in upward pressure on Dai price, and a lower Dai Savings Rate will reduce this pressure.
+
+The higher the Dai Savings Rate, the greater the incentive to deposit DAI and earn interest. However, removing Dai from the markets may lower liquidity in certain liquidity 
```
Again, try to be really clear as to what the trade-offs are. Increased DAI demand versus higher cost to the protocol. Increased DAI demand versus lowering liquidity.

Try to say what is essential without too much extra. Focus on the trade-offs when increasing versus decreasing, rather than arbitrary high and low values. Basically, to increase the DSR is to trade-off revenue / maybe liquidity for DAI demand (which should increase the value of DAI.) To decrease it is to trade-off DAI demand for revenue, however, this might be a pure good if peg is high. 

Also maybe comment on growth as well. It's not just the effect on the peg. It makes DAI more attractive in general as an asset - which makes it easier to expand to new markets, convince users to use DAI instead of competitors, etc.

LongForWisdom commented on governance/parameter-docs/param-dai-savings-rate.md:
```diff
@@ -0,0 +1,53 @@
+
+The higher the Dai Savings Rate, the greater the incentive to deposit DAI and earn interest. However, removing Dai from the markets may lower liquidity in certain liquidity 
+pools, leading to increased slippage when trades occur. Similarly, it is advisable to keep the Dai Savings Rate lower than vault stability fees to ensure that
+The Maker Protocol maintains a positive cash flow.
```
Line break shouldn't be here.

LongForWisdom commented on governance/parameter-docs/param-dai-savings-rate.md:
```diff
@@ -0,0 +1,53 @@
+pools, leading to increased slippage when trades occur. Similarly, it is advisable to keep the Dai Savings Rate lower than vault stability fees to ensure that
+The Maker Protocol maintains a positive cash flow.
+
+If Dai remains consistently above the Peg, it would seem ill-advised to maintain a high Dai Savings Rate as this will prevent the price from returning to the Peg.
```
Probably not the best place for this. Should go in the Changes section as: 'If the peg is breaking high, this might be something to adjust downward.'

LongForWisdom commented on governance/parameter-docs/param-dai-savings-rate.md:
```diff
@@ -0,0 +1,53 @@
+
+**Why increase this parameter?**
+
+The primary reason for increasing the Dai Savings Rate is to encourage Dai holders to lock their Dai in the contract - thus removing it from the market. Raising the Dai Savings 
```
Probably makes sense to go a step below this. The primary reason is to increase DAI demand, which may have positive affects on the DAI peg, or on DAI adoption generally.

LongForWisdom replied:
Or maybe it would be a step above. I guess what I'm saying is that the end goal is unlikely to be 'make the DSR contract full' because that isn't something governance wants by itself, it wants the affect a full DSR contract has (which is less DAI in the market, which is higher value of DAI)

LongForWisdom commented on governance/parameter-docs/param-dai-savings-rate.md:
```diff
@@ -0,0 +1,53 @@
+**Why increase this parameter?**
+
+The primary reason for increasing the Dai Savings Rate is to encourage Dai holders to lock their Dai in the contract - thus removing it from the market. Raising the Dai Savings 
+Rate may be considered if Dai is trading below the Peg.
```
Line breaks. Also feels maybe less necessary once the above comment is taken into account.

LongForWisdom commented on governance/parameter-docs/param-dai-savings-rate.md:
```diff
@@ -0,0 +1,53 @@
+## Changes
+Adjusting the Dai Savings Rate parameter is a manual process that requires an executive vote. Changes to the Dai Savings Rate are subject to the GSM Pause Delay.
+
+**Why increase this parameter?**
```
Other reasons too? Increase adoption, etc. More attractive to stablecoin users. Better competitively, etc. Encourages integrations with DAI?

LongForWisdom commented on governance/parameter-docs/param-dai-savings-rate.md:
```diff
@@ -0,0 +1,53 @@
+
+**Why decrease this parameter?**
+
+Decreasing the Dai Savings Rate should result in less Dai locked in the contract and increase Dai's supply on the market. In addition, lowering the Dai Savings Rate might be 
```
Try to make the first sentence a direct answer to the question posed in the header. "The primary reason to decrease the DSR is..."

LongForWisdom commented on governance/parameter-docs/param-dai-savings-rate.md:
```diff
@@ -0,0 +1,53 @@
+Decreasing the Dai Savings Rate should result in less Dai locked in the contract and increase Dai's supply on the market. In addition, lowering the Dai Savings Rate might be 
+considered if Dai trades above the Peg to reduce price pressure upwards.
+
+Maker Governance may also wish to decrease the Dai Savings Rate when The Maker Protocol experiences a drop in revenue. If Maker Governance does not reduce the Dai Savings Rate 
```
This is good. Though again, fix line breaks :)

LongForWisdom commented on governance/parameter-docs/param-dai-savings-rate.md:
```diff
@@ -0,0 +1,53 @@
+when income falls, the Maker Protocol could potentially encounter negative cash flows. Negative cash flow would reduce the System Surplus Buffer, and if necessary,
+The Maker Protocol may need to mint MKR to cover any shortfall.
+
+## Considerations
```
Couple of things worth noting directly:
* Nothing stops governance from paying more out in the DSR than it gets in fees. Doing this can decrease the buffer and eventually print MKR (you mentioned this elsewhere but best to call it out explicitly here as well.)
* Talk about how this interacts with stablecoin PSMs. This is moderately important, as the PSM holds the peg in the opposite direction to the DSR.
* Make it clear that because this can't go below zero, it is bounded and can't help decrease demand if it's already set at zero. (I didn't phrase that well, but you get it.)

LongForWisdom replied:
Also, maybe triple check with PE that the lowest value is actually 0%. I'm like 90% sure it is, but it's good to be sure. Kurt and Chris are both super nice.

mkrorbrkr replied:
Checked with Kurt and have changed the text - while it can be set at a negative value it will not actually have any effect as the drip function will revert - therefore DSR balances cannot be decreased

LongForWisdom replied:
Nice, good thing to include in the considerations section.

LongForWisdom commented on governance/parameter-docs/param-dai-savings-rate.md:
```diff
@@ -0,0 +1,53 @@
+When Dai is deposited in the Dai Savings Rate, interest is paid from accrued stability fees. Therefore, increasing the Dai Savings Rate will cause the System Surplus Buffer to
+fill more slowly and will reduce the amount of Dai available for FLAP auctions.
+
+In the event of an Emergency Shutdown, the Dai Savings Rate will be set to 0 to prevent the total debt of the system from increasing.
```
I didn't know this!

LongForWisdom reviewed (changes requested):
Good work so far!

mkrorbrkr commented:
Hey, @LongForWisdom I've updated the DSR document based on your feedback so have kept that here. I've deleted the surplus lot size and will run this by Gala separately but thought given you'd seen the previous draft would make sense to keep that with you. If you'd prefer Gala to look at it first let me know and I'll delete the PR

mkrorbrkr commented on governance/parameter-docs/param-dai-savings-rate.md:
```diff
@@ -9,45 +9,44 @@ Technical Docs: https://docs.makerdao.com/mcd-developer-guides/developer-guides-
 
 ## Description
-The Dai Savings Rate allows Dai holders to deposit Dai into the contract and earn interest paid in Dai. Maker Governance can alter the rate at which interest accrues.
+The Dai Savings Rate is a tool that Maker Governance can use to adjust the interest rate paid to DAI holders who have deposited DAI into the DSR contract. Maker Governance directly controls the Dai Savings rate.
```
Yeah I was trying to make a distinction between the DSR contract (DSR) and the DAI Savings Rate parameter but I agree that this is confusing, will edit.

mkrorbrkr replied:
Think it's easier to change all mentions of the contract to 'Pot' after the first one for clarity

mkrorbrkr commented on governance/parameter-docs/param-dai-savings-rate.md:
```diff
@@ -9,45 +9,44 @@ Technical Docs: https://docs.makerdao.com/mcd-developer-guides/developer-guides-
-Maker Governance can utilise the Dai Savings Rate to encourage Dai holders to deposit Dai to the Dai Savings Rate contract. Higher amounts of Dai within the Dai Savings Rate
-should reduce the market supply of Dai and act as a means to control the Peg.
+
+The purpose of the Dai Savings Rate is to enable Maker Governance to encourage DAI holders to deposit DAI within the DSR contract. Higher amounts of DAI within the Dai Savings Rate should reduce the market supply of DAI and act as upwards pressure on the DAI peg.
```
Agree this is clearer. Edited

mkrorbrkr commented on governance/parameter-docs/param-dai-savings-rate.md:
```diff
@@ -9,45 +9,44 @@ Technical Docs: https://docs.makerdao.com/mcd-developer-guides/developer-guides-
-The Dai Savings rate is usually expressed as a percentage and can be any value greater than 0%. Therefore, the DAI Savings rate cannot be negative.
-
-The Dai Savings Rate can also be turned off by Maker Governance - rendering the interest rate in effect to be 0%.
+The Dai Savings Rate is usually expressed as a percentage and can be set to any value. However, while Governance can set the Dai Savings Rate to a negative value, this will not affect the balances held within the DSR as they cannot decrease; this is because the `drip` function will revert if negative.
```
While I agree this isn't as clear as it could be and have changed the wording based on your advice, I was of the understanding that we are aiming this documentation at Governance participants rather than the general public. At least that's what I think @LongForWisdom told me. Happy to be corrected if wrong

mkrorbrkr commented on governance/parameter-docs/param-dai-savings-rate.md:
```diff
@@ -9,45 +9,44 @@ Technical Docs: https://docs.makerdao.com/mcd-developer-guides/developer-guides-
 ## Considerations
-When Dai is deposited in the Dai Savings Rate, interest is paid from accrued stability fees. Therefore, increasing the Dai Savings Rate will cause the System Surplus Buffer to
-fill more slowly and will reduce the amount of Dai available for FLAP auctions.
+When DAI holders deposit DAI to the DSR contract, interest is paid from accrued stability fees. Therefore, increasing the Dai Savings Rate will cause the System Surplus Buffer to fill more slowly and reduce the amount of Dai available for FLAP auctions. If the Dai Savings Rate is set too high, the Maker Protocol could have negative cash flow and eventually need to print MKR.
```
updated
I think some of the already published parameter docs haven't got this consistently done but we can look at that once we've finished the ones we are working on

LongForWisdom commented on governance/parameter-docs/param-dai-savings-rate.md:
```diff
@@ -0,0 +1,50 @@
+## Description
+The Dai Savings Rate is a tool that Maker Governance can use to adjust the interest rate paid to DAI holders who have deposited DAI into the Dai Savings Rate contract (`Pot`). Maker Governance directly controls the Dai Savings Rate.
+
+The Dai Savings Rate is usually expressed as a percentage and can be set to any value. However, even though Maker Governance can set the Dai Savings Rate to a negative value, this will not affect the balances held within the `Pot`. The balances within the `Pot` will not decrease because the drip function will revert if negative.
```
Feels like this goes into a little too much detail for this section. The key information here is that:
* It's expressed as a percentage as an APR/APY (I forget which of these is correct)
* It's valid for any positive value.

LongForWisdom replied:
Maybe give an idiot-proof example here as well. If the DSR is set to 1%, and a DAI Holder has locked 100 DAI in the DSR for the entire year, they should expect to have 101 DAI after a year. (Check that these numbers are correct though, because they depend on whether it's compounding, which I _think_ it is.)

mkrorbrkr replied:
Noted and updated. It's an APY and is a compounding rate per: https://docs.makerdao.com/smart-contract-modules/rates-module#dai-savings-rate-accumulation

LongForWisdom commented on governance/parameter-docs/param-dai-savings-rate.md:
```diff
@@ -0,0 +1,50 @@
+
+## Purpose
+
+The Dai Savings Rate aims to enable Maker Governance to encourage DAI holders to deposit DAI within the `Pot`. Higher amounts of DAI within the `Pot` should reduce the market supply of DAI and act as upwards pressure on the DAI peg.
```
Maybe 'aims to enable' -> 'allows'. Also I would avoid the use of `pot` and exchange for 'Dai Savings Rate contract' everywhere except for the initial description section. Like, we want people to know what the technical name for it is, but we should use the more human readable name, imo.

mkrorbrkr replied:
Updated - it's a little confusing that the parameter and the contract are both Dai Savings Rate so wasn't sure the best way to differentiate but have just specified whenever I'm talking about the contract

LongForWisdom commented on governance/parameter-docs/param-dai-savings-rate.md:
```diff
@@ -0,0 +1,50 @@
+
+## Trade-offs
+
+Increasing the Dai Savings Rate is a trade-off between increased DAI demand and decreased DAI liquidity and may also result in increased expenses for the Maker Protocol.
```
There's no 'may' here really. The only case in which this does not increase expenses for the protocol is if there is exactly zero DAI locked in the DSR.

LongForWisdom replied:
Also, try to be super clear about which way the trade-off goes. Increasing the Dai Savings Rate increases DAI demand at the cost of increasing protocol expenses, and decreasing DAI liquidity.

LongForWisdom commented on governance/parameter-docs/param-dai-savings-rate.md:
```diff
@@ -0,0 +1,50 @@
+
+Increasing the Dai Savings Rate is a trade-off between increased DAI demand and decreased DAI liquidity and may also result in increased expenses for the Maker Protocol.
+
+Decreasing the DAI Savings Rate will decrease the expenses of the Maker Protocol but may simultaneously reduce DAI demand.
```
Maybe expand a little here? Decreasing the rate may cause DAI holders to move their DAI out of the savings rate contract, potentially increasing the market supply of DAI. Or something.

LongForWisdom commented on governance/parameter-docs/param-dai-savings-rate.md:
```diff
@@ -0,0 +1,50 @@
+
+The primary reason for increasing the Dai Savings Rate is to increase DAI demand. This increase in demand is achieved by encouraging DAI holders to deposit DAI to the `Pot` - thus removing it from the market, which should create upwards pressure on the Peg. Therefore, raising the Dai Savings Rate should be considered if DAI is trading below the Peg.
+
+Increasing the Dai Savings Rate should make DAI a more attractive asset for holders than other stablecoin assets and help integration with the Maker Protocol, for example, if other DAPPs with stablecoin holdings deposit to the `Pot` to earn interest.
```
maybe 'incentivize additional integrations with DAI and the Maker Protocol'?

LongForWisdom commented on governance/parameter-docs/param-dai-savings-rate.md:
```diff
@@ -0,0 +1,50 @@
+
+The primary reason to decrease the Dai Savings Rate is to reduce DAI demand. Reducing the Dai Savings Rate should result in less DAI deposited in the `Pot` and increase market supply. In addition, it may be advisable to decrease the Dai Savings Rate if DAI trades above the Peg to reduce upwards price pressure.
+
+Maker Governance may also wish to decrease the Dai Savings Rate when The Maker Protocol experiences a drop in revenue. If Maker Governance does not reduce the Dai Savings Rate when income falls, the Maker Protocol could encounter negative cash flows. Negative cash flow would reduce the System Surplus Buffer, and if necessary, The Maker Protocol may need to mint MKR to cover any shortfall.
```
Need to be clearer about when MKR is minted, maybe.

"Negative cash flow would reduce the System Surplus Buffer, and if necessary, The Maker Protocol may need to mint MKR to cover any shortfall." -> "Negative cash flow would reduce the System Surplus Buffer, and if the buffer is depleted, the Protocol will mint and sell MKR to cover the shortfall."

LongForWisdom commented on governance/parameter-docs/param-dai-savings-rate.md:
```diff
@@ -0,0 +1,50 @@
+
+Maker Governance should recognise that a negative interest rate will not work as intended as balances within the `Pot` cannot decrease.
+
+Any changes to the Dai Savings Rate may affect PSM usage behaviour. For example, increased upward pressure on the Peg caused by the Dai Savings Rate may require increased PSM usage to balance.
```
Maybe note that this can potentially lead to negative cash flows quite easily, as there is no ongoing income from the PSM, but there is ongoing outflows from the DSR.

LongForWisdom commented on governance/parameter-docs/param-dai-savings-rate.md:
```diff
@@ -0,0 +1,50 @@
+
+Any changes to the Dai Savings Rate may affect PSM usage behaviour. For example, increased upward pressure on the Peg caused by the Dai Savings Rate may require increased PSM usage to balance.
+
+In the event of an Emergency Shutdown, the Dai Savings Rate will be set to 0% to prevent the total debt of the Maker Protocol from increasing.
```
Is this automatic? Nice!

mkrorbrkr replied:
Yep! Very cool

LongForWisdom reviewed (changes requested):
Definitely getting closer @mkrorbrkr. Thanks for your continued work on it :)

mkrorbrkr commented:
added RWA Agreement to the PR as requested by @Gala-G

mkrorbrkr commented:
Updated DSR docs as per @LongForWisdom feedback
